incubator/hnc/internal/validators/hncconfig_test.go

```diff
@@ -165,17 +166,45 @@ func TestNonRBACTypes(t *testing.T) {
 
 func TestPropagateConflict(t *testing.T) {
 	tests := []struct {
-		name         string
-		inNamespaces string
-		allow        bool
+		name            string
+		forest          string
+		inNsesWithAnnot map[string]map[string]string
+		allow           bool
+		errContain      string
 	}{{
-		name:         "Objects with the same name existing in namespaces that one is not an ancestor of the other would not cause overwriting conflict",
-		inNamespaces: "bc",
-		allow:        true,
+		name:   "Objects with the same name existing in namespaces that one is not an ancestor of the other would not cause overwriting conflict",
+		forest: "-aa",
+		inNsesWithAnnot: map[string]map[string]string{
+			"b": map[string]string{},
+			"c": map[string]string{},
+		},
+		allow: true,
+	}, {
+		name:   "Objects with the same name existing in namespaces that one is an ancestor of the other would have overwriting conflict",
+		forest: "-aa",
+		inNsesWithAnnot: map[string]map[string]string{
+			"a": map[string]string{},
+			"b": map[string]string{},
+		},
+		allow: false,
 	}, {
-		name:         "Objects with the same name existing in namespaces that one is an ancestor of the other would have overwriting conflict",
-		inNamespaces: "ab",
-		allow:        false,
+		name:   "Should not cause a conflict if the object in the parent namespace has an exceptions selector that choose not to propagate to the conflicting child namespace",
+		forest: "-aa",
+		inNsesWithAnnot: map[string]map[string]string{
+			"a": map[string]string{api.AnnotationTreeSelector: "c"},
+			"b": map[string]string{},
+		},
+		allow: true,
+	}, {
+		name:   "Should not cause a conflict if the object in the ancestor namespace has a none exceptions selector",
```

Comment:
You're testing a lot more than that. If this were all you were testing, you'd just need a->b with the `none` selector in `a`. But instead you have `a->b->c` with the none selector in `a` but an conflicting object in `b`. Can you find a better way to describe this?

Reply:
Changed to `Should identify the real conflicting source when there are multiple conflicting sources but only one gets propagated`, does this sound ok?